server/http_hooks.go

```diff
@@ -47,127 +43,38 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 	// Check to make sure you're still in the team
 	if teamMember, err := p.API.GetTeamMember(action.Context.TeamID, action.Context.UserID); err != nil || teamMember == nil || teamMember.DeleteAt > 0 {
 		p.API.LogError("Didn't have access to team", "user_id", action.Context.UserID, "team_id", action.Context.TeamID)
-		encodeEphermalMessage(w, "WelcomeBot Error: You do not appear to have access to this team")
+		p.encodeEphermalMessage(w, "WelcomeBot Error: You do not appear to have access to this team")
 		return
 	}
 
 	switch r.URL.Path {
 	case "/addchannels":
-
 		for _, wm := range p.getWelcomeMessages() {
-
 			if data.Team.Name == wm.TeamName {
 				for _, ac := range wm.Actions {
 					if ac.ActionName == action.Context.Action {
 						p.processActionMessage(*data, action, *ac)
-						encodeEphermalMessage(w, "")
+						p.encodeEphermalMessage(w, "")
```

Comment:
I'm a little confused by `""`, seems unnecessary but ok since it's not a change.

Reply:
It's not. You have to return valid JSON to action requests. Otherwise the server will through an error. Hence, this is needed. (But there are more elegant ways to write this)